src/lib.rs

```diff
@@ -1,5 +1,16 @@
-pub mod bot;
+#![feature(result_map_or)]
```

Comment:
I don't think we should require nightly just for a single function. It would be easier to either create an extension trait or free function that provides this functionality.